PyInstaller/building/makespec.py

```diff
@@ -399,6 +401,11 @@ def main(scripts, name=None, onefile=None,
     if DEBUG_ALL_CHOICE[0] in debug:
         debug = DEBUG_ARGUMENT_CHOICES
 
+    # If file extension of main script is '.pyw', force --windowed option.
+    # scripts[0] is a Path object.
+    if os.path.splitext(scripts[0].path)[-1] == '.pyw':
```

Comment:
Strings have not `path` attribute. The script names are simple strings, no py.path objects.

Reply:
Another possibility would be to move makespec.py lines 374-377 to right after the '.pyw' check and before the dictionary `d = {'scripts': scripts, ...}` is created (so line ~408).  `scripts` as a list of `Path` objects is not used before it is stored in `d`.

Reply:
@htgoebel I moved the '.pyw' check to above where `scripts` is converted to a list of `Path` objects so that `scripts[0]` is a string.

Reply:
Fine. Thanks :-)